homeassistant/components/climate/netatmo.py

```diff
@@ -0,0 +1,148 @@
+"""
+Support for Netatmo thermostat.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/thermostat.netatmo/
+"""
+import logging
+from datetime import timedelta
+import requests
+import voluptuous as vol
+
+from homeassistant.components.netatmo import ThermostatData
+from homeassistant.const import TEMP_CELSIUS, ATTR_TEMPERATURE
+from homeassistant.components.climate import (STATE_HEAT, STATE_IDLE, ClimateDevice, PLATFORM_SCHEMA)
+from homeassistant.util import Throttle
+from homeassistant.loader import get_component
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['netatmo']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_RELAY = 'relay'
+CONF_THERMOSTAT = 'thermostat'
+CONF_AWAY_TEMPERATURE = 'away_temperature'
+
+DEFAULT_AWAY_TEMPERATURE = 14
+# # Return cached results if last scan was less then this time ago
+# # NetAtmo Data is uploaded to server every hour
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_RELAY): cv.string,
+    vol.Optional(CONF_THERMOSTAT, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+def setup_platform(hass, config, add_callback_devices, discovery_info=None):
+    """Setup the NetAtmo Thermostat."""
+    netatmo = get_component('netatmo')
+    device = config.get(CONF_RELAY)
+    away_temp = config.get(CONF_AWAY_TEMPERATURE)
+    import lnetatmo
+    try:
+        data = ThermostatData(netatmo.NETATMO_AUTH, device)
+    except lnetatmo.NoDevice:
+        return None
+
+    for module_name in data.get_module_names():
+        if CONF_THERMOSTAT in config:
+            if config[CONF_THERMOSTAT] != [] and \
+               module_name not in config[CONF_THERMOSTAT]:
+                continue
+
+        add_callback_devices([NetatmoThermostat(data, module_name)])
+
+
+class NetatmoThermostat(ClimateDevice):
+    """Representation a Netatmo thermostat."""
+
+    def __init__(self, data, module_name, away_temp=None):
+        """Initialize the sensor."""
+        self._data = data
+        self._state = None
+        self._name = module_name
+        self._target_temperature = None
+        self._away_temp = away_temp
```

Comment:
I've commented it now.
The Netatmo Thermostat only pushes updates to the web every hour, so when you turn on away, the target temperature is updated after one hour.
When i can extract the away temp off the zones and timetable from the request, this will be used.